python/ray/data/dataset.py

```diff
@@ -1197,14 +1202,16 @@ def filter(
             >>> ds = ray.data.range(100)
             >>> ds.filter(lambda row: row["id"] % 2 == 0).take_all()
             [{'id': 0}, {'id': 2}, {'id': 4}, ...]
```

Comment:
Maybe remove this since we don't want people using the `fn` parameter?

Reply:
Given expr is limited, I thought, we can retain it. Let me remove this one.